### PR TITLE
Feat/devengage 1237

### PR DESCRIPTION
**Pull request by dginty4**

### CODE REVIEW

File: genesyscloud/resource_genesyscloud_outbound_callabletimeset.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+		var sdkCampaigntimeslot platformclientv2.Campaigntimeslot
+
+		campaigntimeslotMap := configcampaigntimeslot.(map[string]interface{})
+		if startTime := campaigntimeslotMap["start_time"].(string); startTime != "" {
+			sdkCampaigntimeslot.StartTime = &startTime
+			//*sdkCampaigntimeslot.StartTime += ".000"
+		}
+		if stopTime := campaigntimeslotMap["stop_time"].(string); stopTime != "" {
+			sdkCampaigntimeslot.StopTime = &stopTime
+			//*sdkCampaigntimeslot.StopTime += ".000"
```
Comment (charliecon) [inline genesyscloud/resource_genesyscloud_outbound_callabletimeset.go:247]: Some leftover comments here as well

File: genesyscloud/data_source_genesyscloud_outbound_callabletimeset_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+						timeZone,
+						generateTimeSlotsBlock("07:00:00.000", "18:00:00.000", "3"),
+						generateTimeSlotsBlock("09:30:00.000", "22:30:00.000", "5"),
+					),
+				) + generateOutboundCallabletimesetDataSource(
+					dataSourceId,
+					callabeTimesetName,
+					"genesyscloud_outbound_callabletimeset."+resourceId,
+				),
+				Check: resource.ComposeTestCheckFunc(),
```
Comment (charliecon) [inline genesyscloud/data_source_genesyscloud_outbound_callabletimeset_test.go:35]: think you forgot to check the ids here

File: docs/resources/outbound_callabletimeset.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## Example Usage
+
+```terraform
+resource "genesyscloud_outbound_callabletimeset" "example-callable-time-set" {
+  name = "Example Callable time set"
+  callable_times {
+    time_zone_id = "Africa/Abidjan"
+    time_slots {
+      start_time = "07:00:00"
```
Comment (charliecon) [inline docs/resources/outbound_callabletimeset.md:29]: Did you say you needed to add the milliseconds to fix that `plan not empty` error? If so, it'd be worth reflecting that in the example.

File: genesyscloud/data_source_genesyscloud_outbound_callabletimeset.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		const pageSize = 100
+
+		callableTimesets, _, getErr := outboundAPI.GetOutboundCallabletimesets(pageSize, pageNum, false, "", "", "", "")
+		if getErr != nil {
+			return resource.NonRetryableError(fmt.Errorf("error requesting callable timeset %s: %s", name, getErr))
+		}
+		if callableTimesets.Entities == nil || len(*callableTimesets.Entities) == 0 {
+			return resource.RetryableError(fmt.Errorf("no callable timeset found with name %s", name))
+		}
+		callableTimeset := (*callableTimesets.Entities)[0]
```
Comment (ronanwatkins) [inline genesyscloud/data_source_genesyscloud_outbound_callabletimeset.go:43]: You can only do this when the GET call takes a `name` parameter. You'll need to have a for loop here and paginate through until you find one that matches the name. Have a look at a few other data sources and you'll find the pattern.

File: genesyscloud/resource_genesyscloud_outbound_callabletimeset.go [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+				Type:        schema.TypeString,
+			},
+			`stop_time`: {
+				Description: `The end time of the interval as an ISO-8601 string, i.e. HH:mm:ss`,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			`day`: {
+				Description: `The day of the interval. Valid values: [1-7], representing Monday through Sunday`,
+				Required:    true,
```
Comment (ronanwatkins) [inline genesyscloud/resource_genesyscloud_outbound_callabletimeset.go:45]: You could add a `ValidateFunc: validation.IntInSlice(),` here to check if it's 1,2,3 etc

File: genesyscloud/resource_genesyscloud_outbound_callabletimeset.go [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+				Type:        schema.TypeString,
+			},
+		},
+	}
+	outboundcallabletimesetcampaigntimeslotResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			`start_time`: {
+				Description: `The start time of the interval as an ISO-8601 string, i.e. HH:mm:ss`,
+				Required:    true,
```
Comment (ronanwatkins) [inline genesyscloud/resource_genesyscloud_outbound_callabletimeset.go:35]: For these, a `ValidateDiagFunc: validateTime` function would be nice. If you have a look at `validateDate` you'll see how you could validate a `HH:mm:ss` format

**Comment by ronanwatkins** [comment]
The SDK version has been updated to 80 in another PR that was merged earlier. You'll need to rebase off the latest main and manually change any references to `v75` in your go files to `v80`

**Review by charliecon — APPROVED** [review]

File: docs/resources/outbound_callabletimeset.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
+---
```
Comment (ronanwatkins) [inline docs/resources/outbound_callabletimeset.md:1]: Why was the docs/data-sources/outbound_callanalysisresponseset.md file deleted? ^^

File: genesyscloud/data_source_genesyscloud_outbound_callabletimeset.go [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	sdkConfig := m.(*providerMeta).ClientConfig
+	outboundAPI := platformclientv2.NewOutboundApiWithConfig(sdkConfig)
+
+	timesetName := d.Get("name").(string)
+
+	return withRetries(ctx, 15*time.Second, func() *resource.RetryError {
+		for pageNum := 1; ; pageNum++ {
+			const pageSize = 100
+
+			timesets, _, getErr := outboundAPI.GetOutboundCallabletimesets(pageSize, pageNum, false, "", "", "", "")
```
Comment (ronanwatkins) [inline genesyscloud/data_source_genesyscloud_outbound_callabletimeset.go:37]: Might be safer to have the boolean value as `true`

File: genesyscloud/provider.go [hunk 9]
```diff
@@ -157,10 +158,10 @@ func New(version string) func() *schema.Provider {
 				"genesyscloud_journey_segment":                             dataSourceJourneySegment(),
 				"genesyscloud_location":                                    dataSourceLocation(),
 				"genesyscloud_oauth_client":                                dataSourceOAuthClient(),
-				"genesyscloud_outbound_callanalysisresponseset":            dataSourceOutboundCallAnalysisResponseSet(),
```
Comment (ronanwatkins) [inline genesyscloud/provider.go:160]: Why was this deleted?

**Review by ronanwatkins — APPROVED** [review]
